compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -17762,15 +17762,29 @@ TR::Node *fmaxminSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
          fmin = fmax = secondChild->getFloat();
       else
          {
-         if (firstChild->getFloat() <= secondChild->getFloat())
+         float first = firstChild->getFloat();
+         float second = secondChild->getFloat();
+         int nZeroBits = 0x80000000;
+         if (first <= second)
             {
-            fmin = firstChild->getFloat();
-            fmax = secondChild->getFloat();
+            if (first == second && first == 0)
+               {
+               int firstBits = *(int *)(&first);
```

Comment:
Can we use reinterpret_cast here?

Reply:
actually, I believe its not guaranteed that `float` is 32 bits or, more importantly, that `int` and `float` are the same size right (same for `double` and `long` or `long long`)? Do we have an `int`  type that's `typedefed` or `#defined` so that its the same size as `float`, or do I have to add some more logic to work with the proper size?

Reply:
You did change the reinterpret_cast to use int32_t right, that would use 4 bytes for int32_t. I think it is fair to assume that `float` will be binary floating point usually following `IEEE-754 binary32 format`. You can also call `node->getFloatBits` that would return bits representation of floating point value

Reply:
Also I believe similar simplification for Postive and Negative floats were done in `fdivSimplifier` in same file, you can take a look at that as well to see how special case negative / positive zero is coded.